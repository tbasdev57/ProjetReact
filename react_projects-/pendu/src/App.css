body {
  font-family: "Segoe UI", sans-serif;
  color: #000;
  text-align: center;
}
h1 {
  text-align: center;
}
h1,
input,
button {
  margin-bottom: 10px;
}
.pendu {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
}
.pendu .word {
  width: 100%;
  font: 400 30px/60px 'Segoe UI', sans-serif;
  text-align: center;
}
.pendu .content {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
}
.pendu .content .word {
  letter-spacing: 10px;
}
.pendu .display {
  margin-bottom: 10px;
}
.pendu form {
  display: flex;
  flex-flow: column wrap;
}
.pendu form > * {
  margin-bottom: 10px;
}
.pendu form label > input {
  margin-left: 10px;
}
/*
.App-logo {
	height: 40vmin;
	pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
	.App-logo {
		animation: App-logo-spin infinite 20s linear;
	}
}

.App-header {
	background-color: #282c34;
	min-height: 100vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	font-size: calc(10px + 2vmin);
	color: white;
}

.App-link {
	color: #61dafb;
}

@keyframes App-logo-spin {
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
}
*/
/*# sourceMappingURL=App.css.map */