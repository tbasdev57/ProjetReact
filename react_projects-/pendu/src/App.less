
body{
	font-family: "Segoe UI", sans-serif;
	color: #000;
	text-align: center;
}

h1{text-align: center}

h1, input, button{
	margin-bottom: 10px;
}

.pendu {
	display: flex;
	flex-flow: column wrap;
	justify-content: center;
	.word{
		width: 100%;
		font: 400 30px/60px 'Segoe UI', sans-serif;
		text-align: center;
	}
	.content {
		display: flex;
		flex-flow: row wrap;
		justify-content: center;
		.word{
			letter-spacing: 10px;
		}
	}
	.display {margin-bottom: 10px;}
	form {
		display: flex;
		flex-flow: column wrap;
		> * {
			margin-bottom: 10px;
		}
		label > input {margin-left: 10px}
	}
}


/*
.App-logo {
	height: 40vmin;
	pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
	.App-logo {
		animation: App-logo-spin infinite 20s linear;
	}
}

.App-header {
	background-color: #282c34;
	min-height: 100vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	font-size: calc(10px + 2vmin);
	color: white;
}

.App-link {
	color: #61dafb;
}

@keyframes App-logo-spin {
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
}
*/