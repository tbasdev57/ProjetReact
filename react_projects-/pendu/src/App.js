import React, {Component} from 'react'
import './App.css'

import Letter from "./components/Letter"
import Button from "./components/Button"
import Pendu from "./components/Pendu"
import HallOfFame from "./components/HallOfFame"
import HighScoreInput from "./components/HighScoreInput"

class App extends Component {

    state = {
        word: '',
        gameState: 'start',
        alphabet: 'abcdefghijklmnopqrstuvwxyz'.toUpperCase().split(''),
        checkedLetters: [],
        hallOfFame: null,
        step: 0,
        clicks: 0
    }

    playLetter = (letter) => {
        const {word, checkedLetters, step, gameState, clicks} = this.state
        if (!checkedLetters.includes(letter) && gameState === 'play') {
            if (word.includes(letter)) {
                this.setState({
                    checkedLetters: [...checkedLetters, letter]
                })
            } else {
                this.setState({
                    checkedLetters: [...checkedLetters, letter],
                    step: step + 1
                })
            }
            // Fix for update problems
            if (step+1 === 10) {
                this.setState({gameState: 'perdu'})
            } else if (!this.computeDisplay([...checkedLetters, letter]).includes('_')) {
                this.setState({gameState: 'gagné'})
            }
        }
        this.setState({clicks: clicks+1})
    }

    getStatusLetter(letter) {
        const {checkedLetters, gameState} = this.state
        if (checkedLetters.includes(letter) || gameState !== "play") {
            return 'clicked'
        }
        return 'not_clicked'
    }

    setWord = event => {
        this.setState({word: event.target.value.toUpperCase()})
    }

    computeDisplay(checkedLetters) {
        checkedLetters = typeof checkedLetters !== 'undefined' ? checkedLetters : this.state.checkedLetters;
        const {word, gameState} = this.state
        if(gameState!=="play"){
            return word
        }
        return word.replace(/\w/g, (letter) => (checkedLetters.includes(letter) ? letter : '_'))
    };

    playGame = () => {
        const {word} = this.state
        if (word.includes('_') || word.length === 0) {
            alert("Le mot choisit est invalide, veuillez le modifier")
            return
        }
        this.setState({
            gameState: 'play'
        })
    }

    // Arrow fx for binding
    displayHallOfFame = (hallOfFame) => {
        this.setState({hallOfFame})
    }

    render() {
        const {alphabet, gameState, hallOfFame, clicks, step} = this.state
        return (
            <div className={`pendu ${gameState}`}>
                <h1>Jeu du pendu</h1>
                {gameState === 'start' && (
                    <div className="content">
                        <form className="wordToFind" onSubmit={this.playGame}>
                            <p>Insérer le texte à faire deviner :</p>
                            <input type="text" value={this.state.word} onChange={this.setWord}/>
                            <button type="submit">Démarrer</button>
                        </form>
                    </div>
                )}
                {gameState !== 'start' && (
                    <div className="content">
                        <div className="word">{this.computeDisplay()}</div>
                        {alphabet.map((letter, index) => (
                            <Letter
                                letter={letter}
                                status={this.getStatusLetter(letter)}
                                index={index}
                                key={index}
                                onClick={this.playLetter}
                            />
                        ))}
                    </div>
                )}
                {gameState !== 'start' && (
                    <div className="display">
                        <p>Nombre d'essais restants : {10 - step}</p>
                        <Pendu width="300" height="400" step={step}/>
                    </div>
                )}
                {(gameState !== 'start' && gameState !== 'play') && (
                    <div className="result">
                        {hallOfFame ? (
                            <React.Fragment>
                                <HallOfFame entries={hallOfFame}/>
                                <Button button="Rejouer" onClick={() => window.location.reload(false)}/>
                            </React.Fragment>
                        ) : (
                            <React.Fragment>
                                <div className="word">Vous avez {gameState} !</div>
                                {gameState==="gagné" ? (
                                    <React.Fragment>
                                        <HighScoreInput guesses={clicks} onStored={this.displayHallOfFame} />
                                    </React.Fragment>
                                    ) : (
                                    <Button button="Réessayer" onClick={() => window.location.reload(false)}/>
                                )}
                            </React.Fragment>
                        )}
                    </div>
                )}
            </div>
        )
    }
}

export default App
