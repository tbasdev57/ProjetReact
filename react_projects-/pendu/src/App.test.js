import { expect } from 'chai'
import React from 'react'
import { shallow } from 'enzyme'

import App from './App'

describe('<App />', () => {
    it('renders without crashing', () => {
        shallow(<App />)
    })

    it('has 26 letter', () => {
        const wrapper = shallow(<App />)
        wrapper.setState({gameState: "play"})
        expect(wrapper.find('Letter')).to.have.length(26)
        expect(wrapper).to.matchSnapshot()
    })

    it('should match its reference snapshot', () => {
        const wrapper = shallow(<App />)

        expect(wrapper).to.matchSnapshot()
    })

})