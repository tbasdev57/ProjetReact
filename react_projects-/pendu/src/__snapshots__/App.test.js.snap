// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 26 letter 1`] = `
<div
  className="pendu play"
>
  <h1>
    Jeu du pendu
  </h1>
  <div
    className="content"
  >
    <div
      className="word"
    />
    <Letter
      index={0}
      key="0"
      letter="A"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={1}
      key="1"
      letter="B"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={2}
      key="2"
      letter="C"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={3}
      key="3"
      letter="D"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={4}
      key="4"
      letter="E"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={5}
      key="5"
      letter="F"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={6}
      key="6"
      letter="G"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={7}
      key="7"
      letter="H"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={8}
      key="8"
      letter="I"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={9}
      key="9"
      letter="J"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={10}
      key="10"
      letter="K"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={11}
      key="11"
      letter="L"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={12}
      key="12"
      letter="M"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={13}
      key="13"
      letter="N"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={14}
      key="14"
      letter="O"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={15}
      key="15"
      letter="P"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={16}
      key="16"
      letter="Q"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={17}
      key="17"
      letter="R"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={18}
      key="18"
      letter="S"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={19}
      key="19"
      letter="T"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={20}
      key="20"
      letter="U"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={21}
      key="21"
      letter="V"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={22}
      key="22"
      letter="W"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={23}
      key="23"
      letter="X"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={24}
      key="24"
      letter="Y"
      onClick={[Function]}
      status="not_clicked"
    />
    <Letter
      index={25}
      key="25"
      letter="Z"
      onClick={[Function]}
      status="not_clicked"
    />
  </div>
  <div
    className="display"
  >
    <p>
      Nombre d'essais restants : 
      10
    </p>
    <Pendu
      height="400"
      step={0}
      width="300"
    />
  </div>
</div>
`;

exports[`<App /> should match its reference snapshot 1`] = `
<div
  className="pendu start"
>
  <h1>
    Jeu du pendu
  </h1>
  <div
    className="content"
  >
    <form
      className="wordToFind"
      onSubmit={[Function]}
    >
      <p>
        Insérer le texte à faire deviner :
      </p>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Démarrer
      </button>
    </form>
  </div>
</div>
`;
